addons/level_block/level_block_node.gd

```diff
@@ -81,34 +76,34 @@ func refresh():
 	mesh = create_mesh()
 	mesh.surface_set_material(0, material)
 	material.albedo_texture = texture_sheet
-	# VisualServer
-	visual = VisualServer.instance_create()
-	VisualServer.instance_set_base(visual, mesh)
+	# RenderingServer
+	visual = RenderingServer.instance_create()
+	RenderingServer.instance_set_base(visual, mesh)
 	if is_inside_tree():
-		VisualServer.instance_set_scenario(visual, get_world().scenario)
-		VisualServer.instance_set_transform(visual, global_transform)
-	# PhysicsServer
+		RenderingServer.instance_set_scenario(visual, get_world_3d().scenario)
+		RenderingServer.instance_set_transform(visual, global_transform)
+	# PhysicsServer3D
 	if !generate_collision:
 		return
-	body = PhysicsServer.body_create(PhysicsServer.BODY_MODE_STATIC)
-	PhysicsServer.body_set_mode(body, PhysicsServer.BODY_MODE_STATIC)
-	shape = PhysicsServer.shape_create(PhysicsServer.SHAPE_CONCAVE_POLYGON)
-	PhysicsServer.shape_set_data(shape, mesh.get_faces())
+	body = PhysicsServer3D.body_create()
+	PhysicsServer3D.body_set_mode(body, PhysicsServer3D.BODY_MODE_STATIC)
+	shape = PhysicsServer3D.area_create()
+	PhysicsServer3D.shape_set_data(shape, mesh.get_faces())
```

Comment:
The shape generation has changed a bit in Godot 4. The API is not fully documented, but this is how this should work, I believe.